code/modules/antagonists/revolution/revolution.dm

```diff
@@ -441,56 +438,108 @@
+			<b>You have been chosen out of your fellow provocateurs to rename the station. Choose wisely.</b> Message ends.\""))
+	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_REVOLUTION_TAX_REMOVAL)
 
-			player_mind.add_antag_datum(/datum/antagonist/enemy_of_the_revolution)
+	for (var/mob/living/player as anything in GLOB.player_list)
+		var/datum/mind/player_mind = player.mind
 
-			if (!istype(player))
-				continue
+		if (isnull(player_mind))
+			continue
 
-			if(player_mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
-				ADD_TRAIT(player, TRAIT_DEFIB_BLACKLISTED, REF(src))
-				player.med_hud_set_status()
+		if (!(player_mind.assigned_role.departments_bitflags & (DEPARTMENT_BITFLAG_SECURITY|DEPARTMENT_BITFLAG_COMMAND)))
+			continue
 
-		for(var/datum/job/job as anything in SSjob.joinable_occupations)
-			if(!(job.departments_bitflags & (DEPARTMENT_BITFLAG_SECURITY|DEPARTMENT_BITFLAG_COMMAND)))
-				continue
-			job.allow_bureaucratic_error = FALSE
-			job.total_positions = 0
+		if (player_mind in ex_revs + ex_headrevs)
+			continue
+
+		player_mind.add_antag_datum(/datum/antagonist/enemy_of_the_revolution)
+
+		if (!istype(player))
+			continue
+
+		if(player_mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
+			ADD_TRAIT(player, TRAIT_DEFIB_BLACKLISTED, REF(src))
+			player.med_hud_set_status()
+
+	for(var/datum/job/job as anything in SSjob.joinable_occupations)
+		if(!(job.departments_bitflags & (DEPARTMENT_BITFLAG_SECURITY|DEPARTMENT_BITFLAG_COMMAND)))
+			continue
+		job.allow_bureaucratic_error = FALSE
+		job.total_positions = 0
+
+	var/datum/game_mode/dynamic/dynamic = SSticker.mode
+	dynamic.unfavorable_situation()
+
+	var/message_header = "A recent assessment of your station has marked your station as a severe risk area for high ranking Nanotrasen officials."
+	var/extra_detail = try_auto_call_shuttle() \
+		? "For the safety of our staff, we are expediting an emergency shuttle for remaining members of security and command." \
+		: "For the safety of our staff, we have blacklisted your station for new employment of security and command."
+	var/propaganda = pick(world.file2list("strings/anti_union_propaganda.txt"))
 
-		if (revs_win_injection_amount)
-			var/datum/game_mode/dynamic/dynamic = SSticker.mode
-			dynamic.unfavorable_situation()
+	priority_announce(
+		"[message_header]\n\n[extra_detail]\n\n[propaganda]",
+		sender_override = "Central Command Loyalty Monitoring Division"
+	)
+
+/// How much of the station, ignoring sec and command, should be revs before a shuttle will be automatically called?
+#define REV_AUTO_CALL_THRESHOLD 0.65
+
+/datum/team/revolution/proc/try_auto_call_shuttle()
+	var/total_revs = 0
+	var/total_candidates = 0
+
+	for (var/mob/player as anything in GLOB.player_list)
+		if (player.mind.has_antag_datum(/datum/antagonist/enemy_of_the_revolution))
+			continue
+
+		if (player.mind in ex_revs + ex_headrevs)
```

Comment:
this isn't hot code, right?
additionally, why do we need to check this in the first place?

Reply:
1. no
2. to count the number of revs that were on the station, because they'll have just had their antag datums removed

Reply:
is there any reason we can't sum the length of the two lists and use that?

Reply:
no

Reply:
I mean like... `total_revs = length(ex_revs) + length(ex_headrevs)`

Reply:
Oh right I guess